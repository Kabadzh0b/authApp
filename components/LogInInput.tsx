import {StyleSheet, Text, TextInput, View} from "react-native";
import React from "react";


interface LogInInputProps {
    valid: boolean,
    setValid: (val: boolean) => void,
    width: number,
    value: string,
    setValue: (val: string) => void,
    img: "email" | "password"
}

const LogInInput = ({valid, setValid, value, setValue, width, img}: LogInInputProps) => {
    return (
        <View style={[styles.inputContainer, {width: width}, valid ? {} : {borderColor: "red"}]}>
            <Text>{img === "email" ? "✉️" : "🔒"}</Text>
            <TextInput
                style={styles.input}
                onChangeText={(e) => {
                    setValue(e);
                    setValid(true);
                }}
                value={value}
                placeholder={"Пароль"}
            />
        </View>
    )
}

export default LogInInput;
const styles = StyleSheet.create({
    inputContainer: {
        flexDirection: "row",
        alignItems: "center",
        borderBottomWidth: 1,
        marginBottom: 16,
        maxWidth: 500,
    },
    input: {
        width: '100%',
        height: 40,
        paddingHorizontal: 8,
    },
});
