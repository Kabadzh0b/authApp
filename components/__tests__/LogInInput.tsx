import React from "react";
import renderer from "react-test-renderer";
import LogInInput from "../LogInInput";

it('default email renders correctly', () => {
    const input = renderer
        .create(<LogInInput valid={true} setValid={() => {
        }} width={200} value={"ESFS"}
                            setValue={() => {
                            }} img={"email"}/>)
        .toJSON();
    expect(input).toMatchSnapshot();
})

it('default password renders correctly', () => {
    const input = renderer
        .create(<LogInInput valid={true} setValid={() => {
        }} width={200} value={"ESFS"}
                            setValue={() => {
                            }} img={"email"}/>)
        .toJSON();
    expect(input).toMatchSnapshot();
})

it('empty renders correctly', () => {
    const input = renderer
        .create(<LogInInput valid={true} setValid={() => {
        }} width={200} value={""}
                            setValue={() => {
                            }} img={"email"}/>)
        .toJSON();
    expect(input).toMatchSnapshot();
})

it('valid false renders correctly', () => {
    const input = renderer
        .create(<LogInInput valid={false} setValid={() => {
        }} width={200} value={""} setValue={() => {}} img={"email"}/>)
        .toJSON();
    expect(input).toMatchSnapshot();
})

it('big width renders correctly', () => {
    const input = renderer
        .create(<LogInInput valid={false} setValid={() => {
        }} width={1000} value={""} setValue={() => {}} img={"email"}/>)
        .toJSON();
    expect(input).toMatchSnapshot();
});
