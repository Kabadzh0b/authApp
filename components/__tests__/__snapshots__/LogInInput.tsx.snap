// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`big width renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "maxWidth": 500,
      },
      {
        "width": 1000,
      },
      {
        "borderColor": "red",
      },
    ]
  }
>
  <Text>
    ✉️
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Пароль"
    style={
      {
        "height": 40,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value=""
  />
</View>
`;

exports[`default password renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
      {},
    ]
  }
>
  <Text>
    ✉️
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Пароль"
    style={
      {
        "height": 40,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value="ESFS"
  />
</View>
`;

exports[`default renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
      {},
    ]
  }
>
  <Text>
    ✉️
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Пароль"
    style={
      {
        "height": 40,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value="ESFS"
  />
</View>
`;

exports[`empty renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
      {},
    ]
  }
>
  <Text>
    ✉️
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Пароль"
    style={
      {
        "height": 40,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value=""
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
      {},
    ]
  }
>
  <Text>
    ✉️
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Пароль"
    style={
      {
        "height": 40,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value="ESFS"
  />
</View>
`;

exports[`valid false renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 16,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
      {
        "borderColor": "red",
      },
    ]
  }
>
  <Text>
    ✉️
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Пароль"
    style={
      {
        "height": 40,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value=""
  />
</View>
`;
